src/components/button/index.tsx

```diff
@@ -406,10 +410,12 @@ class Button extends PureComponent<Props, ButtonState> {
   }
 
   getIconStyle() {
-    const {disabled, iconStyle: propsIconStyle, iconOnRight} = this.props;
+    const {disabled, iconStyle: propsIconStyle, iconOnRight, supportRTL} = this.props;
     const size = this.props.size || DEFAULT_SIZE;
+    const shouldFlipRTL = supportRTL && Constants.isRTL;
     const iconStyle: ImageStyle = {
-      tintColor: this.getLabelColor()
+      tintColor: this.getLabelColor(),
+      transform: shouldFlipRTL ? [{scaleX: -1}] : undefined
     };
```

Comment:
The Image already supports this feature, why not just pass forward the `supportRTL` prop